src/main/scala/mesosphere/marathon/api/HTTPMetricsFilter.scala

```diff
@@ -0,0 +1,89 @@
+package mesosphere.marathon
+package api
+
+import javax.servlet._
+import javax.servlet.http.{ HttpServletRequest, HttpServletRequestWrapper, HttpServletResponse, HttpServletResponseWrapper }
+import mesosphere.marathon.metrics.{ Metrics, ServiceMetric }
+
+/**
+  * This filter replaces the default I/O streams with proxies that count
+  * the number of bytes that are sent to and from the client.
+  */
+class HTTPMetricsFilter extends Filter {
+  private[this] val inputBytesMetric = Metrics.counter(ServiceMetric, getClass, "inputBytes")
+  private[this] val outputBytesMetric = Metrics.counter(ServiceMetric, getClass, "outputBytes")
+
+  /**
+    * Wraps a `ServletOutputStream` and overrides the `write` method in
+    * order to count the number of bytes that went through the stream.
+    *
+    * @param upstream The stream to proxy the output to
+    */
+  private class ProxyOutputStream(val upstream: ServletOutputStream) extends ServletOutputStream {
```

Comment:
I would maybe mention the word metrics or Measured or something, proxy is very generic, this is basically a Decorator to collect metrics...

Reply:
True, the term `Proxy` can be confusing 👍